### ISSUE TITLE
Run behind reverse proxy with basic auth

### ISSUE THREAD

**Issue by james-choncholas** [oznu/homebridge-config-ui-x#526]
**Describe The Bug:**
Running behind a reverse proxy that implements basic auth breaks homebridge-ui-x.
UI-X will attach bearer tokens to requests even when it's auth mode is set to "none".
This overwrites the basic auth header (bearer and basic auth use the same header.)

In this configuration, all requests come back 401 - unauthorized (from the reverse proxy) because the basic auth header has been overwritten with a bearer token by UI-X.

**To Reproduce:**
I ran UI-X behind traefik reverse proxy but any reverse proxy with basic auth will work.
The reverse proxy must then be configured to use basic auth (that terminates at the rev proxy.)

**Expected behavior:**
I would expect setting auth mode to "none" disables all authentication and
leaves the Authorization HTTP headers alone.

**Homebridge Config:**

```json
...
    "platforms": [
        {
            "name": "Config",
            "port": 8080,
            "auth": "none",
            "sudo": true,
            "log": {
                "method": "systemd",
                "service": "homebridge"
            },
            "platform": "config"
        }
    ]
...
```

**Environment:**

* **Node.js Version**: v8.17.0 
* **NPM Version**: v6.13.4 
* **Homebridge Version**: 
* **Homebridge Config UI X Version**: v4.10.0 
* **Operating System**: Raspbian
* **Process Supervisor**: Systemd

**Comment by oznu** [comment]
Hi @james-choncholas,

I can see that would be a problem. It's not just a matter of removing the token in the client because the server side still expects a token even in no auth mode.

I'll think about how to solve this one; it will require a reasonable amount of effort to remove the token requirements entirely so I'm not going to promise anything.

**Comment by doug-w** [comment]
Bump, was this meant to be closed because of stale or still be actioned on?  I'm running into it now.

**Comment by Andi1968** [comment]
Hi doug-w, 
I encountered the same problem last week implementing access control with reverse proxy Traefik  and BasicAuth. Additionally I had a problem with Node-Red, it showed continuously "Connection broken, will retry  in .... seconds" although the connection was active. But this affected only Safari Browser. 
Yesterday I switched from BasicAuth to forwardAuth implementing Authelia. 
This solved all my problems and I even have second factor authentication with TOTP (Google Authenticator) and my Yubikey. 
It was really worth the effort.
